src/data/setoid/partition.lean

```diff
@@ -237,6 +244,35 @@ _ (subtype (@is_partition α)) _ (partial_order.to_preorder _) $ partition.order
 
 end partition
 
+section finpartition
+
+open_locale classical
+
+/-
+-- Inserted in data.finset.lattice.lean, line 1115
+lemma finset.sup_id_eq_sUnion {α} (s : finset (set α)) :
+  s.sup id = ⋃₀(↑s) :=
+finset.sup_id_eq_Sup _
+-/
+
+/-- A finite setoid partition furnishes a finpartition -/
+def finpartition.of_partition {c : finset (set α)}
+  (hc : setoid.is_partition (c : set (set α))) : finpartition (set.univ : set α) :=
+{ parts := c, --  finset (set α)
+  sup_indep := finset.sup_indep_iff_pairwise_disjoint.mpr $ eqv_classes_disjoint hc.2,
+  --  self.parts.sup_indep id
+  sup_parts := c.sup_id_eq_sUnion.trans hc.sUnion_eq_univ,
+  --  self.parts.sup id = (set.univ : set α)
+  not_bot_mem := hc.left --  ⊥ ∉ self.parts
+}
```

Comment:
I don't think the comments are worth anything here, they're just repeating the definition of `finpartition`.

Also, we can name this so that dot notation works:
```suggestion
def setoid.is_partition.finpartition {c : finset (set α)} (hc : setoid.is_partition (c : set (set α))) :
  finpartition (set.univ : set α) :=
{ parts := c,
  sup_indep := finset.sup_indep_iff_pairwise_disjoint.mpr $ eqv_classes_disjoint hc.2,
  sup_parts := c.sup_id_eq_sUnion.trans hc.sUnion_eq_univ,
  not_bot_mem := hc.left }
```

Reply:
comments deleted